modules/services/workspace-service/src/main/java/org/wso2/ballerina/tooling/service/workspace/launcher/LaunchManager.java

```diff
@@ -0,0 +1,262 @@
+
+        // send a message if ballerina home is not set
+        if(null == System.getProperty("ballerina.home") || System.getProperty("ballerina.home").isEmpty()){
+            pushMessageToClient(launchSession, LauncherConstants.ERROR, LauncherConstants.ERROR,
+                    LauncherConstants.INVALID_BAL_PATH_MESSAGE);
+            pushMessageToClient(launchSession, LauncherConstants.ERROR, LauncherConstants.ERROR,
+                    LauncherConstants.SET_BAL_PATH_MESSAGE);
+            return;
+        }
+
+        // construct the command
+        command = System.getProperty("ballerina.home") + File.separator + "bin" + File.separator + "ballerina ";
+
+        if(type == LauncherConstants.ProgramType.RUN) {
+            command = command + " run ";
+        }else{
+            command = command + " service ";
+        }
+
+        command = command + filePath + File.separator + fileName;
+
+        if(debug){
+            port = getFreePort();
+            command = command + "  --ballerina.debug " + port;
+        }
+
+        try {
+            this.program = Runtime.getRuntime().exec(command);
+
+            if(type == LauncherConstants.ProgramType.RUN ){
+                pushMessageToClient(launchSession, LauncherConstants.EXECUTION_STARTED, LauncherConstants.INFO,
+                        String.format(LauncherConstants.RUN_MESSAGE , fileName));
+            } else {
+                pushMessageToClient(launchSession, LauncherConstants.EXECUTION_STARTED, LauncherConstants.INFO,
+                        String.format(LauncherConstants.SERVICE_MESSAGE , fileName));
+            }
+
+            if(debug){
+                MessageDTO debugMessage = new MessageDTO();
+                debugMessage.setCode(LauncherConstants.DEBUG);
+                debugMessage.setPort(port);
+                pushMessageToClient(launchSession, debugMessage);
+            }
+
+            // start a new thread to stream command output.
+            Runnable run = new Runnable() {
+                public void run() {
+                    LaunchManager.this.streamOutput();
+                }
+            };
+            (new Thread(run)).start();
+
+        } catch (IOException e) {
+            pushMessageToClient(launchSession, LauncherConstants.EXIT, LauncherConstants.ERROR, e.getMessage());
+            this.program.destroy();
+        }
+    }
+
+    public void streamOutput() {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(this.program.getInputStream()));
+        String line = "";
+        try {
+            while ((line = reader.readLine())!= null) {
+                pushMessageToClient(launchSession, LauncherConstants.OUTPUT, LauncherConstants.DATA, line);
+            }
+            pushMessageToClient(launchSession, LauncherConstants.EXECUTION_STOPED , LauncherConstants.INFO,
+                    LauncherConstants.END_MESSAGE);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Any reason to use e.printStackTrace here? In above you have used "pushMessageToClient" method and pass the exception there.